src/Microsoft.DotNet.ProjectJsonMigration/MSBuildExtensions.cs

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Build.Evaluation;
+using Microsoft.Build.Construction;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.Cli;
+using System.Linq;
+using System.IO;
+
+namespace Microsoft.DotNet.ProjectJsonMigration
+{
+    public static class MSBuildExtensions
+    {
+        public static IEnumerable<ProjectPropertyElement> PropertiesWithoutConditions(
+            this ProjectRootElement projectRoot)
+        {
+            return projectRoot.Properties
+                .Where(p => p.Condition == string.Empty
+                    && p.AllParents.All(parent => string.IsNullOrEmpty(parent.Condition)));
+        }
+
+        public static IEnumerable<ProjectItemElement> ItemsWithoutConditions(
+            this ProjectRootElement projectRoot)
+        {
+            return projectRoot.Items
+                .Where(p => string.IsNullOrEmpty(p.Condition)
+                            && p.AllParents.All(parent => string.IsNullOrEmpty(parent.Condition)));
+        }
+
+        public static IEnumerable<string> Includes(
+            this ProjectItemElement item)
+        {
+            return item.Include.Equals(string.Empty) ? Enumerable.Empty<string>() : item.Include.Split(';');
+        }
+
+        public static IEnumerable<string> Excludes(
+            this ProjectItemElement item)
+        {
+            return item.Exclude.Equals(string.Empty) ? Enumerable.Empty<string>() :  item.Exclude.Split(';');
+        }
+
+        public static IEnumerable<string> AllConditions(this ProjectElement projectElement)
+        {
+            return new string[] { projectElement.Condition }.Concat(projectElement.AllParents.Select(p=> p.Condition));
+        }
+
+        public static IEnumerable<string> CommonIncludes(this ProjectItemElement item, ProjectItemElement otherItem)
+        {
+            return item.Includes().Intersect(otherItem.Includes());
+        }
+
+        public static void RemoveIncludes(this ProjectItemElement item, IEnumerable<string> includesToRemove)
+        {
+            item.Include = string.Join(";", item.Includes().Except(includesToRemove));
+        }
+
+        public static void AddIncludes(this ProjectItemElement item, IEnumerable<string> includes)
```

Comment:
Yes indeed this is a set union and the intention is to disallow duplicates. I like UnionIncludes, it fits the implementation most closely I feel.